Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OpenVRDeviceManager.cs

```diff
@@ -109,13 +112,47 @@ private bool IsNodeTypeSupported(XRNodeState obj)
                     return true;
             }
         }
-        
+
+        /// <summary>
+        /// Function to guess which type of controller is attached from the Joystick array
+        /// </summary>
+        /// <remarks>Note, Unity now caches the array between runs, so if you have more than one controller attached, this will fail
+        /// TODO: Find a better way?</remarks>
+        /// <returns></returns>
+        private SupportedControllerType GetControllerType()
+        {
+            SupportedControllerType returnControllerType = SupportedControllerType.GenericOpenVR;
+            var controllers = Input.GetJoystickNames();
```

Comment:
You should checkout how I was doing this in the master branch.  It'll probably have some useful tidbits.

Essentually you're going to have to poll the joysticks and manage them yourself.

Reply:
got a link reference?

Reply:
https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/master/Assets/HoloToolkit/Input/Scripts/InputSources/GamePadInputSource.cs

Reply:
and also https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/master/Assets/HoloToolkit/Input/Scripts/InputSources/XboxControllerInputSource.cs